### PR TITLE
refactor rowstyle

### PR DESCRIPTION
**Pull request by daniel17903**

### CODE REVIEW

File: Frontend/cypress/integration/rentals.spec.js [hunk 1]
```diff
@@ -1,32 +1,17 @@
...
-function isBeforeDay(m1, m2) {
-  return millisAtStartOfDay(m1) < millisAtStartOfDay(m2);
-}
-
-function isToday(millis) {
-  return millisAtStartOfDay(millis) === Date.UTC(2020, 0, 1);
-}
-
-function isBeforeToday(millis) {
-  return isBeforeDay(millis, Date.UTC(2020, 0, 1));
```
Comment (skjerns) [inline Frontend/cypress/integration/rentals.spec.js:23]: how does this work?

File: Frontend/src/components/Input/ColorDefs.js [hunk 2]
```diff
@@ -1 +1,12 @@
...
+  HIGHLIGHT_GREEN: "#83eb34",
+  HIGHLIGHT_BLUE: "#2d90e0",
+  HIGHLIGHT_YELLOW: "#f7ef0a",
+
+  RENTAL_RETURNED_TODAY_GREEN: "rgb(214, 252, 208)",
+  RENTAL_LATE_RED: "rgb(240, 200, 200)",
+  RENTAL_TO_RETURN_TODAY_BLUE: "rgb(160, 200, 250)",
+
+  DEFAULT_BACKGROUND: "rgb(255, 255, 255)",
+});
```
Comment (skjerns) [inline Frontend/src/components/Input/ColorDefs.js:12]: nice! good :)

File: Frontend/src/components/Table/Cell.svelte [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+    if (typeof r != "undefined") return (r * 299 + g * 587 + b * 114) / 1000;
+  }
+
+  $: if (brightnessByColor(backgroundColor) < 125) {
+    fontColor = "white"; // adaptive font color for darker highlight
+  }
+</script>
+
+<td style={`background-color: ${backgroundColor}; color: ${fontColor}; height: ${rowHeight}px;`}>
```
Comment (skjerns) [inline Frontend/src/components/Table/Cell.svelte:35]: why did you decide not to have a generalized `style` function as before? Could there not be more style changes that we would want to make in the future?

File: Frontend/src/components/Table/Row.svelte [hunk 4]
```diff
@@ -12,68 +13,26 @@
       return col.display ? col.display(item[col.key]) : item[col.key];
     }
   };
+
+  let cellBackgroundColors = new Array(columns.length).fill("white");
+  $: cellBackgroundColorsFunction(item).then(
+    (newCellBackgroundColors) => (cellBackgroundColors = newCellBackgroundColors)
```
Comment (skjerns) [inline Frontend/src/components/Table/Row.svelte:19]: I guess this allows for async loading of background color?

**Review by skjerns — COMMENTED** [review]
looks good so far! 

The thing with the CONFIG I don't fully grasp, but that is due to not having touched this part of the code ever before. But seems like you just moved it to a different file, which seems reasonable

File: Frontend/src/components/Table/Cell.svelte (hunk 3 above)
Comment (daniel17903) [reply Frontend/src/components/Table/Cell.svelte:35]: I think in order to keep the code clean it's better to implement a more generalized style function when we need it.

File: Frontend/cypress/integration/rentals.spec.js (hunk 1 above)
Comment (daniel17903) [reply Frontend/cypress/integration/rentals.spec.js:23]: These functions were used to detect whether a time (given in ms) is at a certain day. Therefore millisAtStartOfDay calculates the time in ms at the start of a day (00:00). Then these start times can be compared to each other in order to detect if both are at the same day or not.

File: Frontend/src/components/Table/Row.svelte (hunk 4 above)
Comment (daniel17903) [reply Frontend/src/components/Table/Row.svelte:19]: Yes exactly. It basically does the same as the {await} in the HTML part but requires less duplicated code in this case.

**Comment by daniel17903** [comment]
> looks good so far!
> 
> The thing with the CONFIG I don't fully grasp, but that is due to not having touched this part of the code ever before. But seems like you just moved it to a different file, which seems reasonable

Yes, basically I just moved this code out of TableEditor. Because TableEditor was quite large and this code is just some kind of configuration. It specifies which database / columns / filters / ... to use for each table.

**Pull request merged**